### PR TITLE
MaterialLoader: Honor missing material properties.

### PR DESCRIPTION
**Pull request by 1993heqiang**
Related PR: #22028

**Description**

Add missing attributes, such as fog, blendDst, linecap, etc.

### CODE REVIEW

**Comment by 1993heqiang** [comment]
@mrdoob Done. 😄

File: src/materials/Material.js [hunk 1]
```diff
@@ -294,6 +303,29 @@ class Material extends EventDispatcher {
 		data.stencilZFail = this.stencilZFail;
 		data.stencilZPass = this.stencilZPass;
 
+		if ( this.clippingPlanes !== null ) data.clippingPlanes = this.clippingPlanes;
```
Comment (Mugen87) [inline src/materials/Material.js:306]: This property is an array of `THREE.Plane`. You can serialize it like that.

Comment (1993heqiang) [reply src/materials/Material.js]: @Mugen87  Already fixed. Thanks. 👍 
image

File: src/loaders/MaterialLoader.js [hunk 2]
```diff
@@ -151,8 +152,55 @@ class MaterialLoader extends Loader {
...
+		if ( json.blendEquation !== undefined ) material.blendEquation = json.blendEquation;
+		if ( json.blendSrcAlpha !== undefined ) material.blendSrcAlpha = json.blendSrcAlpha;
+		if ( json.blendDstAlpha !== undefined ) material.blendDstAlpha = json.blendDstAlpha;
+		if ( json.blendEquationAlpha !== undefined ) material.blendEquationAlpha = json.blendEquationAlpha;
+		if ( json.depthFunc !== undefined ) material.depthFunc = json.depthFunc;
+
+		if ( json.clippingPlanes !== undefined ) {
+
+			const len = json.clippingPlanes.length;
+			material.clippingPlanes = new Array( len );
```
Comment (Mugen87) [inline src/loaders/MaterialLoader.js:166]: Just do:
```js
material.clippingPlanes = [];
```

File: src/materials/Material.js [hunk 3]
```diff
@@ -294,6 +303,44 @@ class Material extends EventDispatcher {
 		data.stencilZFail = this.stencilZFail;
 		data.stencilZPass = this.stencilZPass;
 
+		if ( this.clippingPlanes !== null && Array.isArray( this.clippingPlanes ) ) {
+
+			const len = this.clippingPlanes.length;
+			data.clippingPlanes = new Array( len );
```
Comment (Mugen87) [inline src/materials/Material.js:309]: Same here.

Comment (1993heqiang) [reply src/materials/Material.js]: Solved.

File: src/loaders/MaterialLoader.js (hunk 2 above)
Comment (1993heqiang) [reply src/loaders/MaterialLoader.js]: Solved

File: src/materials/Material.js [hunk 4]
```diff
@@ -294,6 +303,44 @@ class Material extends EventDispatcher {
 		data.stencilZFail = this.stencilZFail;
 		data.stencilZPass = this.stencilZPass;
 
+		if ( this.clippingPlanes !== null && Array.isArray( this.clippingPlanes ) ) {
```
Comment (Mugen87) [inline src/materials/Material.js:306]: It's sufficient to write:
```js
if ( Array.isArray( this.clippingPlanes ) === true ) {
```
`isArray()` covers the `null` case.

Comment (1993heqiang) [reply src/materials/Material.js]: Ok, done. Are there any other reviews?

**Review by Mugen87 — APPROVED** [review]

**Comment by LeviPesin** [comment]
Why this PR was closed?

**Comment by Mugen87** [comment]
It seems there is no demand for serializing/deserializing _all_ material properties. We can probably focus on those which are supported in the editor.

**Pull request closed without merging**